sql/core/src/test/scala/org/apache/spark/sql/JoinSuite.scala

```diff
@@ -200,6 +200,30 @@ class JoinSuite extends QueryTest with SharedSQLContext {
       Nil)
   }
 
+  test("inner join, propagate empty relation before checking Cartesian products") {
```

Comment:
It make make the test more concise if you just test the various join in a single test.